### PR TITLE
[6/N][multi quorum ejection] Create handler and put them all together

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
This is part of the work to compute operator nonsigning rate for ejection in multi quorum.

This is the last step to actually compute the multi quorum nonsigner!

Tested: query result

> {"meta":{"size":50},"data":[{"operator_id":"0c2758536c85f72596ef5f78a64d65cc24f0961ec22c956db6883ff53835dd6c","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"1426175cdf0919375d7a5559024c3b64694f9c8aa3591a83a053f1d5bf5419d4","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"1e6047379920c2982df90a53c1bb1f8013d06c01ae0a632aff0516e3743819c4","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"26d7dc2472359806c14eaa0704159522c4cd251c98e28cd12cb3778473714d76","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"284451714040ff7172be6a03f9611f4691283f1f1913c3e0320a0dceb24e07bf","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"2a6499336addf1c3f8c7f40a847942796fd4b9dcb2589f5f37a9fddad77221bd","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"2d3f0d61933d8620aee38d186b9e17ac0c238a42b6b26162e91adfa7cfc08c21","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"2da1f3a9cba27c7a7f4b8e9880a8e8fcf2c2e1d4c8db1a3052c9456d015972a3","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"37ec536ed77e74df929aa308dedb8078845d8ec3689ad68a65994a002bbcd07f","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"399a7826d328d6a330ec317755566276e428996aed34dd30b4f795a178260e8c","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"3b330c87b11ae8257451ea3fa6300c8552b6031fa57372dea3ecf3d81ac3ec5b","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"3dddd53339d03e060e8e1a566feb43daa2fbc83e7cd54c01cc97594563b4e819","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"45d0fd964be0b21dca37f68a9f732399435feca2d1378b81782eeaef6f581d00","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"4e7afdbc367a65291accb268cdb01f8f826eb90c53385d53c3f7dbc94d76e6bc","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"4f3037cdb331f3c3f60810dd66f7682902d0c9ee14e8ababc18fc850ac07873c","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"50a30064c297bbeb0b2934de27f678b2b491221b0b1af9ac7890267c38f79651","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"50adde5d236c181f9fdc5eee7eb49f1761dada49a6206a2992ddc5c5f1262336","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"66673ad95053510fca73600cc3b8f013d046a8c3f7e7fd572e44767f9ef49459","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"6db901d981a07bdd7133ae37c7e716dcc47abb38403320c3510a620ed0ea2d94","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"71685f1a54be28e419a9b3aeecfb1b8db654c6fe9b41cc00fb1c517c3c7205da","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"74fb3a53f8ea2c0ad92b212c5023bb159a9d81dfc6c81f1da85de104f06f921b","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"7999718fd8b714a43c5edc4eff5c2b161f142512577fd21a7c5f9c233f84c1fe","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"7d33a56a822beccbad260d3b59a4566efd4a6e9cc1c466d74ab44a465468a131","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"7fbd3e3555f8eb8ccf09613e41911ec9c6ca42dcf5f202274648b35adcca78b9","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"94eb2c8d20da7f410c88899ed14675fccfec54e114c9330c08fd127b078f2e71","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"984cf64a479011aa63423337bc285ad7a9e7db9aeaed0823575f114113b7c249","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"9c21197890a9e0b4202d4b3601aac583b24426b85d246ff1199ca49c961a4c85","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"a8ed40d08e4390b4c2f5f8050379750db1874550f2aa4dad0fed1964c1706ada","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"b2fbbcf956835cd35ba685b7dfb9bd4b896de72f9e8e3bc7f4ff9f03ce439899","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"b869c30cdaa7f03c1443429e67b59dc65ec15c74b49daf3a373f22fef9c0f646","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"ba26efbb6d3173920b489a6bb2f0ab2b1bdc9296cedf5299c68c8d65b4bdd71f","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"bc09a41bd145c97070db4ae784b6f41e6153a9e9c289e1ff054566e77c3b45a2","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"bdd4d72a247c12658c64ae65e93139705fc6b786b329cd8021027b49012b9d0c","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"c8a289d0cc1a5c472f6bdf80931bc84e2fd367cdc3f11482a58a8b54e5936489","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"dcc6e7d9f62f4a7024a88a2c651a5470e19d70888dd07c1861cb7598471e9c9b","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"e02eb821060cd7b07b2f998d5409117190a38200eff09700ffaa322c5819008d","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"e484bac7e9d948e479214f80ef3efd8777f7c7555655e88a661110dcea8046d7","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"f3cbbd02a20044c58ed57c8dfee9fd5d70858c3d3f449051651f0d5649031ac1","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"f4981ec373c0a01d3c55c3278c779a3c5df935d887419be1dad23bbc58a3e97b","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"f879e6698bee1baf2d1f9197dab8f86d807a9abb129687cdad3beb65be1136b1","quorum_id":0,"total_unsigned_batches":27,"total_batches":27,"percentage":100},{"operator_id":"ad61768cf8a0557916323486213d15f56351e89c1cc91030a0b18b45e650f73a","quorum_id":0,"total_unsigned_batches":12,"total_batches":27,"percentage":44.44},{"operator_id":"e4ef41861087510abb0a4ec824bf3f055c6fcb5afc2a8e644ac51817c1ae3af3","quorum_id":0,"total_unsigned_batches":7,"total_batches":27,"percentage":25.93},{"operator_id":"228275bb077d0b961e851b7e6688ee9c7f196ba5230f0c401c3669073fb514de","quorum_id":0,"total_unsigned_batches":3,"total_batches":27,"percentage":11.11},{"operator_id":"7c19d1af6fa14b444c1faff3fd31676113b93f42972b75766da79a5bcd544f95","quorum_id":0,"total_unsigned_batches":2,"total_batches":27,"percentage":7.41},{"operator_id":"11285884611d6638e1dd2bc41a98be60daa7169b3e8082833192fa1fc361da4e","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7},{"operator_id":"3d6669da200305720356ae7436bd6c3cdf2b9ed0aa5942d309d61d9cc5570979","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7},{"operator_id":"5b99ff3ad362280058cad0855a10eef8898b27a47ae1307c3e02b2a1276d8dd6","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7},{"operator_id":"7162f8254c5e087ae8b484a9483332972c52432e5a46273d2a265b2f91997204","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7},{"operator_id":"7da45adc02c0448f698fb510bda21dfee8cc79c690a882db198165ce9398b98e","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7},{"operator_id":"9b5f0928733acff9b05529575e6eacb413acdbceb743904416e8c4de641acfe6","quorum_id":0,"total_unsigned_batches":1,"total_batches":27,"percentage":3.7}]}

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: core/eth/tx.go [hunk 1]
```diff
@@ -591,19 +591,20 @@ func (t *Transactor) GetQuorumBitmapForOperatorsAtBlockNumber(ctx context.Contex
...
-		index  uint32
+		index  int
 		err    error
 	}
 	resultChan := make(chan BitmapOrError, len(indices))
 	pool := workerpool.New(maxNumWorkerPoolThreads)
 	for i, index := range indices {
-		idx := index
+		i := i
+		index := index
```
Comment (ian-shim) [inline core/eth/tx.go:601]: `i`, `index` are pretty confusing.
Can we call `indices` something like `quorumIndices` and leave a comment saying each element in `quorumIndices` corresponds to the quorum bitmap index of each operator in `byteOperatorIds` in the same order?

File: disperser/dataapi/nonsigner_handler.go [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+	// failed to sign for operator "op" and quorum "q".
+	numFailed := computeNumFailed(batches, operatorQuorumIntervals)
+
+	// Compute num batches responsible, where numResponsible[op][q] is the number of batches
+	// that operator "op" and quorum "q" are responsible for.
+	numResponsible := computeNumResponsible(batches, operatorQuorumIntervals)
+
+	// Compute the nonsigning rate for each <operator, quorum> pair.
+	operators := make([]*OperatorNonsigningPercentageMetrics, 0)
+	for op, val := range numResponsible {
```
Comment (ian-shim) [inline disperser/dataapi/nonsigner_handler.go:74]: nit: val doesn't mean anything. Can we name it something like numResponsibleByQuorum?

File: disperser/dataapi/nonsigner_handler.go [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+	// Compute num batches failed, where numFailed[op][q] is the number of batches
+	// failed to sign for operator "op" and quorum "q".
+	numFailed := computeNumFailed(batches, operatorQuorumIntervals)
+
+	// Compute num batches responsible, where numResponsible[op][q] is the number of batches
+	// that operator "op" and quorum "q" are responsible for.
+	numResponsible := computeNumResponsible(batches, operatorQuorumIntervals)
+
+	// Compute the nonsigning rate for each <operator, quorum> pair.
+	operators := make([]*OperatorNonsigningPercentageMetrics, 0)
```
Comment (ian-shim) [inline disperser/dataapi/nonsigner_handler.go:73]: nit: s/operators/nonsignerMetrics

**Review by ian-shim — APPROVED** [review]
lgtm, some comments on naming

File: core/eth/tx.go (hunk 1 above)
Comment (jianoaix) [reply core/eth/tx.go]: done

File: disperser/dataapi/nonsigner_handler.go (hunk 2 above)
Comment (jianoaix) [reply disperser/dataapi/nonsigner_handler.go:74]: done

File: disperser/dataapi/nonsigner_handler.go (hunk 3 above)
Comment (jianoaix) [reply disperser/dataapi/nonsigner_handler.go]: done

**Pull request merged**